### PR TITLE
new perldoc parser

### PR DESCRIPTION
**Pull request by jdorweiler**
Replacement for outdated perldoc fathead 

TODO:
-some descriptions are cut off https://beta.duckduckgo.com/?q=perl+rand&ia=about
-parser for perlvar page

https://duck.co/ia/view/perl_doc

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/fathead/perl_doc/Parsers/Util.pm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+# return tab formatted row for fathead output.txt file
+sub get_row {
+    my ($title, $abs, $url, $type, $redirect) = @_;
+    $type = 'A' unless $type;
+    if($type eq 'A'){
+        return "$title\t$type\t\t\t\t\t\t\t\t\t\t$abs\t$url";
+    }
+    elsif($type eq 'R'){
+        return "$redirect\t$type\t$title\t\t\t\t\t\t\t\t\t\t";
```
Comment (GuiltyDolphin) [inline lib/fathead/perl_doc/Parsers/Util.pm]: How about `. "\t" x (however many tabs that is)`?

File: lib/fathead/perl_doc/Parsers/Util.pm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+}
+
+sub trim_abstract {
+    my($full_abs, $len) = @_;
+    
+    my @split_abs = split /\s/, $full_abs;
+    my @final_abs;
+    if(scalar @split_abs >= ($len-1)){
```
Comment (GuiltyDolphin) [inline lib/fathead/perl_doc/Parsers/Util.pm]: We don't need `scalar` here.

File: lib/fathead/perl_doc/Parsers/Util.pm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        @final_abs = splice(@split_abs, 0, ($len-1));
+        
+        foreach my $word (@split_abs){
+            my $last;
+            
+            if($word =~ /\.$|\?$|\!$/){
```
Comment (GuiltyDolphin) [inline lib/fathead/perl_doc/Parsers/Util.pm]: `/[.?!]$/`

**Comment by jbarrett** [comment]
Continuing in #247 - thanks @jdorweiler!

**Pull request closed without merging**